### DISCUSSION TITLE
Explain how OldestUnackedMessageAge in PubSub works

### DISCUSSION THREAD

**Discussion by kamilglod** [kedacore/keda#3665] [Q&A / Need Help] [answered]
I'm trying to understand how the `OldestUnackedMessageAge` mode works in https://keda.sh/docs/2.8/scalers/gcp-pub-sub/

As I can see in the implementation you're using value from the metric which is a oldest message age in seconds as a metric value. So e.g. when we have a 120s message age we're scaling up to 120 replicas? Or we are adding single replica on each check (which is more desired approach I think)?

**Comment by JorTurFer** [comment, marked as answer]
Hello,
Basically KEDA will query the value of the metric "pubsub.googleapis.com/subscription/oldest_unacked_message_age" and will return that value, so the HPA controller will use it to decide the amount of replicas. If you have your target value in 10 and the metric is 100, the HPA will spawn 10 replicas, if the metric goes to 120, the HPA will add 2 extra replicas, etc
